third-party/aosp-dexutils/src/main/java/com/tencent/tinker/android/dex/Mutf8.java

```diff
@@ -107,7 +107,7 @@ public static void encode(byte[] dst, int offset, String s) {
      * Returns an array containing the <i>modified UTF-8</i> form of {@code s}.
      */
     public static byte[] encode(String s) throws UTFDataFormatException {
-        int utfCount = (int) countBytes(s, true);
+        int utfCount = (int) countBytes(s, false);
```

Comment:
This change (and the one below) is due to an interesting issue regarding String lengths.

With the new Android Gradle Plugin 4.0, the dex classes were suddenly containing some `String` values greater than `65535` in length.  When these functions are set to `true`, `countBytes` throws an error if the length is greater than `65535`.  This length, however, I believe is only a problem in Java, which does not let you store `String` literals in the source code greater than `65535`.

This does not appear to be true in Dex, as the Dex format indicates that the size of a `String` is represented by a `uleb128`, which is unbounded.  Essentially, setting this to false, fixed the issue, and the patch generation succeeded.